File: bin/beginer_training/Day03_4.js
function solution(n) {
	return Array.from({length: n}, (_, i) => i + 1).filter(item => item%2 !== 0); 
}

// Array.from 배열 생성
// {length: n} : 길이가 n 인 배열과 유사한 객체
// (element, index) => ... 
// 길이가 n 인 배열을 만들고 배열의 요소를 1부터 시작하는 숫자로 채움

// for(let i=1; i<=n; i++) {
// 	if(i%2!=0) answer.push(i);
// }
// return answer;

// for(let i=1; i<=n; i+=2) answer.push(i);

File: src/beginer_training/Day03_4.java
package beginer_training;

import java.util.stream.IntStream;

public class Day03_4 {
	public int[] solution(int n){
		return IntStream.rangeClosed(1, n)
		.filter(i -> i%2 !=0)
		.toArray();
	}
}

// 번외 문제풀이
// List<Integer> list = new ArrayList<>();
// for(int i=1; i<=n; i+=2) list.add(i);
// return list.stream().mapToInt(Integer::intValue).toArray();

// IntStream.rangeClosed().filter(i -> i%2 !=0) i 는 스트림 요소의 값
// 메소드 참조 ::
// Integer 객체의 intValue 메소드를 호출해서, Integer 객체를 int 타입으로 변환
